plugin/trino-faker/src/main/java/io/trino/plugin/faker/FakerPageSource.java

```diff
@@ -243,6 +238,107 @@ public void close()
         closed = true;
     }
 
+    private Generator sequenceGenerator(FakerColumnHandle handle, long rowOffset)
+    {
+        SequenceWriter writer = sequenceWriter(handle);
+
+        return new Generator()
+        {
+            long currentRowId = rowOffset;
+
+            @Override
+            public void accept(BlockBuilder blockBuilder)
+            {
+                writer.accept(blockBuilder, currentRowId++);
+            }
+        };
+    }
+
+    private SequenceWriter sequenceWriter(FakerColumnHandle handle)
+    {
+        Range genericRange = handle.domain().getValues().getRanges().getSpan();
+        Type type = handle.type();
+        // check every type in order defined in StandardTypes
```

Comment:
why order matters?

Reply:
Only for consistency, there are now 3 blocks of code that enumerate different types, so I wanted to document why they're in this particular order.